workflow/controller/dag.go

```diff
@@ -314,6 +314,14 @@ func (woc *wfOperationCtx) executeDAGTask(dagCtx *dagContext, taskName string) {
 	node := dagCtx.getTaskNode(taskName)
 	task := dagCtx.GetTask(taskName)
 	if node != nil && node.Fulfilled() {
+		// Collect the completed task metrics
+		tmpl := woc.wf.GetTemplateByName(task.Template)
```

Comment:
Are we sure `task.Template` will always resolve? Even if it's a `templateRef`?

Reply:
What if `task` is  actually an expanded task with a `withItems`?

Reply:
Good catch, we need to cover those usecases